src/status_im2/contexts/chat/events.cljs

```diff
@@ -167,23 +167,26 @@
 (rf/defn close-chat
   {:events [:chat/close]}
   [{:keys [db] :as cofx} navigate-to-shell?]
-  (when-let [chat-id (:current-chat-id db)]
-    (chat.state/reset-visible-item)
-    (rf/merge cofx
-              (merge
-               {:db (dissoc db :current-chat-id)}
-               (let [community-id (get-in db [:chats chat-id :community-id])]
-                 ;; When navigating back from community chat to community, update switcher card
-                 ;; A close chat event is also called while opening any chat.
-                 ;; That might lead to duplicate :dispatch keys in fx/merge, that's why dispatch-n is
-                 ;; used here.
-                 (when (and community-id config/shell-navigation-disabled? (not navigate-to-shell?))
-                   {:dispatch-n [[:shell/add-switcher-card
-                                  :community-overview community-id]]})))
-              (link-preview/reset-all)
-              (delete-for-me/sync-all)
-              (delete-message/send-all)
-              (offload-messages chat-id))))
+  (let [chat-id (:current-chat-id db)
+        view-id (:view-id db)]
+    (when (and chat-id (or config/shell-navigation-disabled? (not= view-id :chat)))
+      (chat.state/reset-visible-item)
+      (rf/merge
+       cofx
+       (merge
+        {:db (dissoc db :current-chat-id)}
+        (let [community-id (get-in db [:chats chat-id :community-id])]
+          ;; When navigating back from community chat to community, update switcher card
+          ;; A close chat event is also called while opening any chat.
+          ;; That might lead to duplicate :dispatch keys in fx/merge, that's why dispatch-n is
+          ;; used here.
```

Comment:
Not directly related, but just adding for context
https://github.com/facebook/react-native/issues/14825
https://stackoverflow.com/questions/44909375/react-native-keyboardavoidingview-not-animated